machine/v1/types_powervsprovider.go

```diff
@@ -0,0 +1,154 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// PowerVSResourceTypeID enum property to identify an ID type resource reference
+	PowerVSResourceTypeID PowerVSResourceType = "ID"
+	// PowerVSResourceTypeName enum property to identify an Name type resource reference
+	PowerVSResourceTypeName PowerVSResourceType = "Name"
+	// PowerVSResourceTypeRegEx enum property to identify a tags type resource reference
+	PowerVSResourceTypeRegEx PowerVSResourceType = "RegEx"
+)
+
+// PowerVSResourceType enum attribute to identify the type of resource reference
+type PowerVSResourceType string
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PowerVSMachineProviderConfig is the type that will be embedded in a Machine.Spec.ProviderSpec field
+// for a PowerVS virtual machine. It is used by the PowerVS machine actuator to create a single Machine.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +k8s:openapi-gen=true
+type PowerVSMachineProviderConfig struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// userDataSecret contains a local reference to a secret that contains the
+	// UserData to apply to the instance.
+	// +optional
+	UserDataSecret *corev1.LocalObjectReference `json:"userDataSecret,omitempty"`
+
+	// credentialsSecret is a reference to the secret with IBM Cloud credentials.
+	// +optional
+	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret,omitempty"`
```

Comment:
We would prefer to use a specific local type for these references, see this convention https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#use-specific-types-for-object-references-and-omit-ref-suffix for more details